crates/zint/cli/src/main.rs

```diff
@@ -0,0 +1,261 @@
+use anyhow::{Context, Result};
+use clap::{Parser, Subcommand};
+use std::fs;
+use zint::utils::{find_up, FoundryConfig};
+use zint::Contract;
+
+#[derive(Parser)]
+#[command(name = "cargo-zint")]
+struct Cli {
+    #[command(subcommand)]
+    command: ZintCommand,
+}
+
+#[derive(Subcommand)]
+enum ZintCommand {
+    /// Zink testing commands
+    Zint {
+        #[command(subcommand)]
+        subcommand: Commands,
+    },
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Create a new ztests crate in the Foundry project
+    New,
+    /// Run the tests in the ztests crate
+    Run,
+}
+
+fn main() -> Result<()> {
+    let cli = Cli::parse();
+
+    match cli.command {
+        ZintCommand::Zint { subcommand } => match subcommand {
+            Commands::New => create_ztests_crate()?,
+            Commands::Run => run_ztests()?,
+        },
+    }
+
+    Ok(())
+}
+
+fn create_ztests_crate() -> Result<()> {
+    // Find foundry.toml in the current directory or parent directories
+    let foundry_toml_path = find_up("foundry.toml")?;
+    let foundry_toml_content =
+        fs::read_to_string(&foundry_toml_path).context("Failed to read foundry.toml")?;
+    let foundry_config: FoundryConfig =
+        toml::from_str(&foundry_toml_content).context("Failed to parse foundry.toml")?;
+
+    // Determine the output directory (default to "out" if not specified)
+    let out_dir = foundry_config
+        .profile
+        .default
+        .out
+        .unwrap_or_else(|| "out".to_string());
+    let _out_path = foundry_toml_path.parent().unwrap().join(&out_dir);
+
+    // Create ztests directory
+    let ztests_path = foundry_toml_path.parent().unwrap().join("ztests");
+    fs::create_dir_all(&ztests_path).context("Failed to create ztests directory")?;
+
+    // Fetch the zink version from the environment variable set by zint-cli's build.rs
+    let zink_version = std::env::var("ZINK_VERSION").unwrap_or_else(|_| "0.1.12".to_string());
+
+    // Write ztests/Cargo.toml with workspace dependencies
+    let cargo_toml_content = format!(
+        r#"
+[package]
+name = "ztests"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+anyhow = "1.0"
+zink = {{ path = "../../../../../zink", version = "{}", features = ["abi-import"] }}
+zint = {{ path = "../../../zint" }}
```

Comment:
```suggestion
zink = "^0.1"
zint = "^0.1"
```

the hard coded dependencies from path is not correct